c/include/libsbp/cpp/state.h

```diff
@@ -37,6 +37,10 @@ class State {
   IReader *reader_;
   IWriter *writer_;
 
+  State(const State&);
```

Comment:
The code you've added doesn't actually achieve the goal of disabling the copying of the class, it just tells the compiler that there should be copy constructor and assignment constructor defined somewhere, in the code. The compiler won't realize that there is an error until it performs the linking state of the build process, which is one of the last steps. The best way to achieve the goal of disabling of a copy constructor/assignment is to do the following.

```
State(const State&) = delete;
State& operator=(const State&) = delete;
```

After you make the change, try and write a unit test to verify that it works (you won't be able to commit the work because it will error out the compiler, but at least it is a way to check that your code works).

Reply:
Agreed. Eventually we might want to enable move operations, but disallowing copies should implicitly disallow moves for now.